app/controllers/api/base_controller.rb

```diff
@@ -38,17 +38,20 @@ def get_resource
 
     # overwrites resource_scope in FindCommon to consider nested objects
     def resource_scope(options = {})
-      options[:association] ||= controller_name
-      if nested_obj && nested_obj.respond_to?(options[:association])
-        association = nested_obj.send(options[:association])
-        if association.respond_to?(:authorized)
-          association.authorized(options[:permission], resource_class)
-        else
-          association
-        end
-      else
-        super(options)
-      end
+      child_scope = super(options)
+      child_scope.merge(parent_scope).readonly(false)
+    end
+
+    def parent_scope(options = {})
+      parent_name, parent_resource_scope = parent_resource_details()
+
+      return resource_class.where(nil) unless parent_resource_scope
```

Comment:
In [Rails 4.0.13](https://github.com/rails/rails/blob/v4.0.13/activerecord/lib/active_record/associations/association.rb#L91) it's only deprecated. In 4.1 we can use `#all`. Although stackoverflow has a lot of good answers, it is not a good source of "right way to do it". `where(nil)` is IMHO bad idea since it's not clear how nil is converted to condition in official documentation and may behave differently in different Rails versions. If you really want to get ready for Rails 4.1+, use `Rails.version`.